### PR TITLE
Add a marker to the map

### LINKED ISSUE #40: Week 4 Project: Maps - Step - 2
Add a marker to the map

### PR DESCRIPTION
**Pull request by SibingWu**
## Description

Add a draggable marker with drop & bouncing effect.

Fixes #40 

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Test
Manual Test

### CODE REVIEW

File: portfolio/src/main/webapp/map.js [hunk 1]
```diff
@@ -1,6 +1,30 @@
...
+      title: "Stan the T-Rex"
+    });
+
+    trexMarker.addListener("click", function() {
+        toggleBounce(trexMarker);
+    });
+}
+
+/**
+ * Generates the bouncing animation for marker.
```
Comment (htthach) [inline portfolio/src/main/webapp/map.js]: I think the comment does not reflect the method?
Doesn't look like we generate animation? Look like we toggle the animation of the marker on / off?

File: portfolio/src/main/webapp/map.js [hunk 2]
```diff
@@ -1,6 +1,30 @@
 /** Creates a map and adds it to the page. */
 function createMap() {
-  const map = new google.maps.Map(
-      document.getElementById('map'),
-      {center: {lat: 37.422, lng: -122.084}, zoom: 16});
+    const map = new google.maps.Map(
+        document.getElementById('map'),
+        {center: {lat: 37.422, lng: -122.084}, zoom: 16});
```
Comment (htthach) [inline portfolio/src/main/webapp/map.js]: Looks like you want to center to trexMarker.position?

File: portfolio/src/main/webapp/map.js (hunk 1 above)
Comment (SibingWu) [reply portfolio/src/main/webapp/map.js]: Fixed

File: portfolio/src/main/webapp/map.js (hunk 2 above)
Comment (SibingWu) [reply portfolio/src/main/webapp/map.js]: I refactor it to extract out as a const variable

Comment (SibingWu) [reply portfolio/src/main/webapp/map.js]: Fixed

**Review by htthach — APPROVED** [review]

**Pull request merged**